### ISSUE TITLE
Prop accessibilityLabel not working as intended on react-native-windows

### ISSUE THREAD

**Issue by YajurG** [react-native-datetimepicker/datetimepicker#803]
# Bug report

## Summary

When using a DateTimePicker object in React Native for Windows, setting accessibilityLabel does not have the expected functionality for accessibility. Setting accessibilityLabel should populate the Name property in the UIA tree with the same value. From looking at the DateTimePicker codebase, accessibilityLabel seems to be included in the WindowsNativeProps object, so having a value for accessibilityLabel should populate the Name property in the UIA tree, but this is not the case. 

## Reproducible sample code

import React from 'react';
import {AppRegistry, View} from 'react-native';
import DateTimePicker from '@react-native-community/datetimepicker';

```
export default class Bootstrap extends React.Component {
  render() {
    return (
      <View
        accessible={true}
        style={{borderRadius: 30, width: 60, height: 60, margin: 10}}>
        <DateTimePicker
          accessibilityLabel="Simple Example"
          value={new Date()}
          onChange={() => {}}
          mode="date"
          style={{width: 200, opacity: 1, height: 50}}
        />
      </View>
    );
  }
}

AppRegistry.registerComponent('Bootstrap', () => Bootstrap);
```

## Steps to reproduce

1. Download Accessibility Insights for Windows
2. Run the code example from above or download React Native Gallery and navigate to the DateTimePicker example page.
3. Run Accessibility Insights for Windows.
4. Hover over the DateTimePicker example and look at the Name property in the UIA Details page on Accessibility Insights.
5. The Name property will be undefined but should match the accessibilityLabel property value.

Screenshot of UIA output:
![image](https://github.com/react-native-datetimepicker/datetimepicker/assets/30995198/a42625d3-2392-4bf1-9ed7-f7312bbf0db3)

Describe what you expected to happen:

The name property in the UIA table on Accessibility Insights should match the accessibilityLabel property value.

## Environment info

`datetimepicker version number: 7.4.1`
`npx react-native info` output:

```bash
System:
  OS: Windows 10 10.0.22621
  CPU: "(24) x64 AMD Ryzen Threadripper PRO 3945WX 12-Cores     "
  Memory: 44.28 GB / 63.86 GB
Binaries:
  Node:
    version: 18.12.1
    path: C:\Program Files\nodejs\node.EXE
  Yarn:
    version: 1.22.19
    path: C:\Program Files (x86)\Yarn\bin\yarn.CMD
  npm:
    version: 8.19.2
    path: C:\Program Files\nodejs\npm.CMD
  Watchman: Not Found
SDKs:
  Android SDK: Not Found
  Windows SDK:
    AllowDevelopmentWithoutDevLicense: Enabled
    AllowAllTrustedApps: Enabled
    Versions:
      - 10.0.18362.0
      - 10.0.19041.0
      - 10.0.22000.0
      - 10.0.22621.0
IDEs:
  Android Studio: AI-213.7172.25.2113.9123335
  Visual Studio:
    - 17.6.33829.357 (Visual Studio Enterprise 2022)
    - 16.11.33801.447 (Visual Studio Enterprise 2019)
Languages:
  Java:
    version: 18.0.2.1
    path: C:\Program Files\Common Files\Oracle\Java\javapath\javac.EXE
  Ruby: Not Found
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.0.0
    wanted: 18.0.0
  react-native:
    installed: 0.72.1
    wanted: ^0.72.0
  react-native-windows:
    installed: 0.72.0
    wanted: 0.72.0
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: false
  newArchEnabled: false
iOS:
  hermesEnabled: Not found
  newArchEnabled: Not found
```
